Import from gerrit

josephlewis42 opened the pull request:
Initial PR for a Logstash Google Cloud Storage input. This has general feature parity with our beat which had general feature parity with the LS S3 plugin. Meaning the following major features:

 * Gzipped files
 * codecs
 * JSON and ADC credentials
 * using attributes to mark files as processed
 * regex file matchers
 * Unique IDs for events that match what `gcsbeat` provides for idempotent inserts

Tagging @rosbo as someone that's reviewed this internally and @erjohnso, @evandbrown as interested parties.

evandbrown commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,7 @@
+All contributions are welcome: ideas, patches, documentation, bug reports, complaints, and even something you drew up on a napkin.
+
+Programming is not a required skill. Whatever you've seen about open source and maintainers or community members  saying "send patches or die" - you will not see that here.
```
"community members  saying" <-- two spaces before "saying". nittiest nit.

josephlewis42 replied:
Nice catch, looks like it's a bug in their upstream template too if you want the fame and glory of fixing it: https://github.com/elastic/logstash/blob/master/README.md

josephlewis42 commented:
Friendly ping for status of this PR :-)

josephlewis42 commented:
Have you had a chance to take a look at this yet @jsvd?

Thanks!

markhuds commented on docs/index.asciidoc:
```diff
@@ -0,0 +1,264 @@
+    bucket_id => "my-logs-bucket"
+    json_key_file => "/home/user/key.json"
+    file_matches => ".*json"
+    codec => "json_lines"
```
Do not get output with this codec, probably this doesn't work because of line.chomp? https://github.com/logstash-plugins/logstash-input-google_cloud_storage/pull/1/files#diff-a10ed0af350ff953f0040542399af1c9R39 -- easy workarounds but I did bash my head around for a bit before I found the problem (just use `codec => "json"` or `codec => "plain"` with a json filter plugin)

karenzone commented:
@josephlewis42  You did it again.  Nice documentation that builds without error right out of the gate. 
If you think that you might want to link to section headings later, please consider defining the anchors for them now using the format that you're already using (`[id="plugins-{type}s-{plugin}-xyz"]`). 

Thanks for this.  It's always a pleasure to work with you.

guyboertje commented:
Apologies about the delay.

I'm concerned about the version of Jackson being loaded. In Logstash we currently use 2.9.8

josephlewis42 commented:
Hi @guyboertje,

Thank you for the review. I updated the JARs being imported for the GCS client and version is now closer to what LS uses.

I've also removed the line chomp and added new anchors to the documentation like @karenzone recommended (although we don't plan on using them any time soon so they can go out at your convenience).

With that, I bumped the version to 0.10.0.

Let me know what you think. I'm more than happy to port this over to straight Java once Java plugins go GA. I'm super excited about that and the classloader isolation work.

Thanks!
\- Joseph

evandbrown commented:
Hi folks, looks like this failed for LOGSTASH_BRANCH=6.0 && 6.x. Any feedback from the logstash folks would be appreciated. Thanks!

guyboertje commented:
@evandbrown 
We did a mass update of the Travis yaml files on the plugins 2 weeks ago. Here is an example
https://github.com/logstash-plugins/logstash-input-kafka/commit/4adfec607d73a6950e8b24996c14744d4883023d
I guess you should make the same change.

josephlewis42 commented:
Alright, Travis is now updated and is passing.

jsvd commented:
A million thanks to all for your code, reviews and patience. Version 0.10.0 has been published https://rubygems.org/gems/logstash-input-google_cloud_storage/versions/0.10.0-java
